pkg/apis/contrail/v1alpha1/base_types.go

```diff
@@ -732,48 +732,28 @@ func PodIPListAndIPMapFromInstance(instanceType string,
 }
 
 // NewCassandraClusterConfiguration gets a struct containing various representations of Cassandra nodes string.
-func NewCassandraClusterConfiguration(name string, namespace string, client client.Client) (*CassandraClusterConfiguration, error) {
+func NewCassandraClusterConfiguration(name string, namespace string, client client.Client) (CassandraClusterConfiguration, error) {
 	var cassandraNodes []string
 	var cassandraCluster CassandraClusterConfiguration
-	var port string
-	var cqlPort string
-	var jmxPort string
 	cassandraInstance := &Cassandra{}
 	err := client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, cassandraInstance)
 	if err != nil {
-		return &cassandraCluster, err
+		return cassandraCluster, err
 	}
 	for _, ip := range cassandraInstance.Status.Nodes {
 		cassandraNodes = append(cassandraNodes, ip)
 	}
-	cassandraConfigInterface := cassandraInstance.ConfigurationParameters()
-	cassandraConfig := cassandraConfigInterface.(CassandraConfiguration)
-	port = strconv.Itoa(*cassandraConfig.Port)
-	cqlPort = strconv.Itoa(*cassandraConfig.CqlPort)
-	jmxPort = strconv.Itoa(*cassandraConfig.JmxLocalPort)
+	cassandraConfig := cassandraInstance.ConfigurationParameters()
+	endpoint := cassandraInstance.Status.ClusterIP + ":" + strconv.Itoa(*cassandraConfig.Port)
```

Comment:
Maybe store just the ClusterIP in the serverIPList instead, it would simplify things?

Reply:
Done